pypeclub/OpenPype
# Tools: New select context dialog

## Description
Have option to show dialog where context is selected by an artist as subprocess and store result into a json file. It should be also possible predefine context to some level (e.g. limit context selected to specific project and asset).

## Changes
- add new tool with `ContextDialog`
- the tool can be launched using new cli command `contextselection`
    - there are few optional arguments
    - `--project <project name>` - to lock project
    - `--asset <asset name>` - to lock asset
    - `--strict` is a flag and tells the dialog that context must be selected and confirmed
- moved few tool related parts from specific tools into utils as are used at multiple places (`ProjectModel`, `ProjectSortFilterProxy`, `TasksModel`, `TasksProxyModel`)

### How to test
- run command `./openpype_console contextselection <path to json>` (or use poetry python)
- if you'll select and confirm context it should save it to the json file
